src/benchmark_runner.cc

```diff
@@ -140,23 +140,102 @@ void RunInThread(const BenchmarkInstance* b, IterationCount iters,
   manager->NotifyThreadComplete();
 }
 
+struct BenchTimeType {
+  enum { ITERS, TIME } tag;
+  union {
+    int iters;
+    double time;
+  };
+};
+
+BenchTimeType ParseBenchMinTime(const std::string& value) {
+  BenchTimeType ret;
+
+  if (value.empty()) {
+    ret.tag = BenchTimeType::TIME;
+    ret.time = 0.0;
+    return ret;
+  }
+
+  if (value.back() == 'x') {
+    std::string num_iters_str = value.substr(0, value.length() - 1);
+    int num_iters = std::atoi(num_iters_str.c_str());
```

Comment:
if we have `strtonum` we can get an error, but i'm not sure we do.  at least `strtol` provides some range protection (atoi does no such thing).